### ISSUE TITLE
combining marks - args and kwargs calculation

### ISSUE THREAD

**Issue by davin-geiger** [pytest-dev/pytest#2897]
If I add the same marker at different code levels (module, class, function) the args of the combined marker will be taken from the "closest" marker and the kwargs from the "farthest" marker.

**Example Test**
``` python
from pytest import mark
pytestmark = mark.test("argument0", Key0="Value0")

@mark.test("argument1", Key0="Value1")
class TestWithClassMarker(object):
    @mark.test("argument2", Key0="Value2")
    def test_marker(self, request):
        test_marker = request.node.get_marker('test')
        test_kwargs = test_marker.kwargs
        test_args = test_marker.args

        assert test_args[0] == "argument2"
        assert test_kwargs['Key0'] == "Value2"
```
**Result**
``` bash
    @mark.test("argument2", Key0="Value2")
    def test_marker(self, request):
        test_marker = request.node.get_marker('test')
        test_kwargs = test_marker.kwargs
        test_args = test_marker.args

        assert test_args[0] == "argument2"
>       assert test_kwargs['Key0'] == "Value2"
E       AssertionError: assert 'Value0' == 'Value2'
E         - Value0
E         ?      ^
E         + Value2
E         ?      ^
```

**Comment by RonnyPfannschmidt** [comment]
thats "expected" behaviour due to mark combination - it can be worked around by iterating the markinfo object to get indivudial marker instances that are noncombined

this is part of the ongoing mark issues
there is a plan to provide a api thats not as broken but it needs time to implement

**Comment by davin-geiger** [comment]
Thanks for the very quick reply. In that case, I will create my own kwargs by iterating through the markinfos of the marker, as you suggested.

**Comment by RonnyPfannschmidt** [comment]
fixed in #3317
